packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2902,6 +2917,13 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         // we may not have seen every sequence number (because of system ops) so apply everything once we
         // don't have any more saved ops
         await this.pendingStateManager.applyStashedOpsAt();
+
+        // If it's not the case, we should take it into account when calculating dirty state.
+        assert(this.context.attachState === AttachState.Attached,
```

Comment:
Isn't it possible for `processSavedOps` to be called when rehydrating a detached container?

Reply:
Today's design assumes on of the two:
1. Detached container: there are no ops at all. We serialize full state (snapshot) and load from it.
2. Attached container: we serialize base snapshot, sequenced ops on top of it and local (stashed) ops.

Check in constructor ensures this API is called only for case # 2 (we have pending state).

If design changes in the future, we will adjust accordingly

Reply:
That makes sense. I did not realize that detached container won't have any stashed ops :-).